sdks/python/apache_beam/io/lyft/s3_and_kinesis.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+import random
+import string
+from collections import namedtuple
+
+from apache_beam import PTransform
+from apache_beam.pvalue import PBegin
+from apache_beam.pvalue import PCollection
+from apache_beam.transforms.core import Windowing
+from apache_beam.transforms.window import GlobalWindows
+
+Event = namedtuple('Event', 'name lateness_in_sec lookback_days')
+KinesisConfig = namedtuple('KinesisConfig', 'name properties parallelism stream_start_mode')
+S3Config = namedtuple('S3Config', 'parallelism lookback_hours')
+
+
+class S3AndKinesisInput(PTransform):
+    """Custom composite transform that uses Kinesis and S3 as
+    input sources. This wraps the streamingplatform-dryft-sdk SourceConnector.
+    Only works with the portable Flink runner.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.events = []
+        self.s3_config = S3Config(None, None)
+        self.source_name = 'S3andKinesis_' + self._get_random_source_name()
+
+    def expand(self, pbegin):
+        assert isinstance(pbegin, PBegin), (
+                'Input to transform must be a PBegin but found %s' % pbegin)
+        return PCollection(pbegin.pipeline)
+
+    def get_windowing(self, inputs):
+        return Windowing(GlobalWindows())
+
+    def with_event(self, event):
+        self.events.append(event)
+        return self
+
+    def with_kinesis_config(self, kinesis_config):
+        self.kinesis_config = kinesis_config
+        return self
+
+    def with_s3_config(self, s3_config):
+        self.s3_config = s3_config
+        return self
+
+    def with_source_name(self, source_name):
+        self.source_name = source_name
+        return self
+
+    @staticmethod
+    @PTransform.register_urn("lyft:flinkS3AndKinesisInput", None)
+    def from_runner_api_parameter(_unused_ptransform, spec_parameter, _unused_context):
+        logging.info("S3AndKinesis spec :" + spec_parameter)
+        instance = S3AndKinesisInput()
+        payload = json.loads(spec_parameter)
+        instance.program_config_path = payload['program_config_path']
```

Comment:
I don't see symmetry between what is defined as part of `to_runner_api_parameter` # json_map and this code.